btree/btree.c

```diff
@@ -1967,23 +2039,24 @@ static const struct seg_ops mem_seg_ops = {
  *  Structure of the node in persistent store.
  */
 struct ff_head {
-	struct m0_format_header ff_fmt;   /*< Node Header */
-	struct node_header      ff_seg;   /*< Node type information */
-	uint16_t                ff_used;  /*< Count of records */
-	uint8_t                 ff_shift; /*< Node size as pow-of-2 */
-	uint8_t                 ff_level; /*< Level in Btree */
-	uint16_t                ff_ksize; /*< Size of key in bytes */
-	uint16_t                ff_vsize; /*< Size of value in bytes */
-	struct m0_format_footer ff_foot;  /*< Node Footer */
+	struct m0_format_header  ff_fmt;    /*< Node Header */
+	struct node_header       ff_seg;    /*< Node type information */
+	uint16_t                 ff_used;   /*< Count of records */
+	uint8_t                  ff_shift;  /*< Node size as pow-of-2 */
+	uint8_t                  ff_level;  /*< Level in Btree */
+	uint16_t                 ff_ksize;  /*< Size of key in bytes */
+	uint16_t                 ff_vsize;  /*< Size of value in bytes */
+	struct m0_format_footer  ff_foot;   /*< Node Footer */
+	void                    *ff_opaque; /*< opaque data */
 	/**
 	 *  This space is used to host the Keys and Values upto the size of the
 	 *  node
 	 */
 } M0_XCA_RECORD M0_XCA_DOMAIN(be);
 
 
-static void ff_init(const struct nd *node, int shift, int ksize, int vsize,
-		    struct m0_be_tx *tx);
+static void ff_init(const struct segaddr *addr, int shift, int ksize, int vsize,
+		    uint32_t ntype, struct m0_be_tx *tx);
 static void ff_fini(const struct nd *node);
```

Comment:
Just like ff_init, can't we take const struct segaddr *addr as parameter in ff_fini as well.

Reply:
I did not touch the part that I did not require. Yash will be taking up this activity to make changes for all the functions.

Reply:
okay